components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OIDCClaimUtil.java

```diff
@@ -0,0 +1,223 @@
+                // Check if requested scope is a supported OIDC scope value
+                if (properties.containsKey(requestedScope)) {
+                    // Requested scope is an registered OIDC scope. Get the claims belonging to the scope.
+                    claimUrisInRequestedScope = getClaimUrisInSupportedOidcScope(properties, requestedScope);
+                    // Iterate the user claims and pick ones that are supported by the OIDC scope.
+                    for (Map.Entry<String, Object> claimMapEntry : userClaims.entrySet()) {
+                        String claimUri = claimMapEntry.getKey();
+                        Object claimValue = claimMapEntry.getValue();
+
+                        if (claimUrisInRequestedScope.contains(claimUri)) {
+                            // User claim is supported by the requested oidc scope.
+                            if (isAddressClaim(claimUri, claimUrisInAddressScope)) {
+                                // Handle Address Claims
+                                populateClaimsForAddressScope(claimUri, claimValue,
+                                        claimUrisInAddressScope, claimValuesForAddressScope);
+                            } else {
+                                returnedClaims.put(claimMapEntry.getKey(), userClaims.get(claimMapEntry.getKey()));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // Some OIDC claims need special formatting etc. These are handled below.
+        handleAddressClaim(returnedClaims, claimValuesForAddressScope);
+        handleUpdateAtClaim(returnedClaims);
+        handlePhoneNumberVerifiedClaim(returnedClaims);
+        handleEmailVerifiedClaim(returnedClaims);
+
+        return returnedClaims;
+    }
+
+
+    private static void populateClaimsForAddressScope(String claimUri,
+                                                      Object claimValue,
+                                                      List<String> addressScopeClaims,
+                                                      Map<String, Object> claimsforAddressScope) {
+        if (claimUri.contains(ADDRESS_PREFIX)) {
+            claimsforAddressScope.put(claimUri.substring(ADDRESS_PREFIX.length()), claimValue);
+        } else if (addressScopeClaims.contains(claimUri)) {
+            claimsforAddressScope.put(claimUri, claimValue);
+        }
+    }
+
+    private static void handleAddressClaim(Map<String, Object> returnedClaims,
+                                           Map<String, Object> claimsforAddressScope) {
+        if (MapUtils.isNotEmpty(claimsforAddressScope)) {
+            final JSONObject jsonObject = new JSONObject();
+            claimsforAddressScope.forEach(jsonObject::put);
+            returnedClaims.put(ADDRESS, jsonObject);
+        }
+    }
+
+    private static List<String> getAddressScopeClaims(Properties oidcProperties) {
+        String[] addressScopeClaims;
```

Comment:
Better use getClaimUrisInSupportedOidcScope internally in method to avoid duplication. Both method use same logic